dev/format_code.sh

```diff
@@ -182,28 +190,27 @@ xmlsort_failed() {
   exit 7
 }
 echo "Formatting resource files with xmlsort..."
-if [ "$ALL" == "--all" ]; then
-  TRACKED=$(git ls-files -- "**.qrc")
-else
-  # Only files which differ from the master branch
-  TRACKED=$(git diff --name-only master -- "**.qrc")
-fi
-UNTRACKED=$(git ls-files --others --exclude-standard -- "**.qrc")
-for file in $TRACKED $UNTRACKED
-do
-  if [ -f "$file" ]; then
-    OLD_CONTENT=$(cat "$file")
-    NEW_CONTENT=$(xmlsort -r "RCC/qresource/file" -i -s "$file" || xmlsort_failed)
-    NEW_CONTENT="${NEW_CONTENT//\'/\"}"
-    if [ "$NEW_CONTENT" != "$OLD_CONTENT" ]
-    then
-      printf "%s\n" "$NEW_CONTENT" > "$file"
-      echo "[M] $file"
-      COUNTER=$((COUNTER+1))
-    else
-      echo "[ ] $file"
-    fi
-  fi
+for file in $(search_files "" .qrc); do
+  xmlsort -r "RCC/qresource/file" -i -s "$file" | tr "'" '"' | update_file "$file" || xmlsort_failed
+done
+
+# Format qml source files with qmlformat
+qml_format_failed() {
+  echo "" >&2
+  echo "ERROR: qmlformat failed!" >&2
+  echo "  Make sure that qmlformat (qtdeclarative5-dev-tools) are installed." >&2
+  echo "  On Linux, you can also run this script in a docker" >&2
+  echo "  container by using the '--docker' argument." >&2
+  exit 7
+}
+echo "Formatting QML sources with qmlformat..."
+for file in $(search_files "" .qml); do
+  ${QMLFORMAT} -l unix "$file" | update_file "$file" || qml_format_failed
 done
 
 echo "Finished: $COUNTER files modified."
+
+# nonzero exit code
+if [ "$CHECK" == ""] && [ $COUNTER -ge 0 ]; then
```

Comment:
Are you sure this is correct? :see_no_evil: It looks like `exit 1` is executed if check mode is **not** enabled (instead of enabled) and if `$COUNTER` is **greater or equal** than zero (which is **always** the case). Or are my Bash skills that bad? :joy: